Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -253,21 +258,14 @@ - (void)deleteTypedCharacter:(unichar)deletedChar withCharacterNowPrecedingCurso
 
 - (void)cursorMovedWithCharacterNowPrecedingCursor:(unichar)c {
     // Determine the type of the character
-    enum CharacterType {
-        CharacterTypeSeparator = 0,
-        CharacterTypeNormal
-    };
-    enum CharacterType currentCharacterType = CharacterTypeNormal;
-    if ([HKWMentionsStartDetectionStateMachine.separatorSet characterIsMember:c] || c == 0) {
-        currentCharacterType = CharacterTypeSeparator;
-    }
+    enum CharacterType currentCharacterType = [self characterTypeOfCharacter:c];
     switch (self.state) {
         case HKWMentionsStartDetectionStateQuiescentReady:
         case HKWMentionsStartDetectionStateQuiescentStalled: {
             // Reset the string buffer
             self.stringBuffer = [@"" mutableCopy];
             self.charactersSinceLastWhitespace = 0;
-            if (currentCharacterType == CharacterTypeSeparator) {
+            if (currentCharacterType == CharacterTypeSeparator || currentCharacterType == CharacterTypeControlCharacter) {
                 // The user moved the cursor to the beginning of the text region, or right after a newline or whitespace or punctuation
                 //  character. This puts the user in the ready state.
                 self.state = HKWMentionsStartDetectionStateQuiescentReady;
```

Comment:
is this read or creating mention, if you've moved your cursor in front of a control char

Reply:
We feel it should be `ready` as mentions is not triggered yet. 
We have now added functionality that once user types any character when preceding character is control character, mentions would be triggered.
For example, user has typed "hello @ world|" ("|" represents cursor). When cursor is moved to character location after '@', state would be set to ready, which means it does not trigger mentions yet and when user starts typing something after '@', we need to modify `characterTyped` method to trigger mentions for the typed text if expected.